From 081386c0d923496c805d1cb3c092f2254d19ceca Mon Sep 17 00:00:00 2001
From: Duanqiang Wen <duanqiangwen@net-swift.com>
Date: Thu, 1 Dec 2022 17:44:36 +0800
Subject: [PATCH] openeuler: net: txgbe: fix arm 32bit arch compile warning.
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 081386c0d923496c805d1cb3c092f2254d19ceca
Modified-by-SEL: No


driver inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I61PSD
CVE: NA

Fix txgbe_main.c warning: right shift count >= width of type warning.

Signed-off-by: Duanqiang Wen <duanqiangwen@net-swift.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/netswift/txgbe/txgbe_main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/netswift/txgbe/txgbe_main.c b/drivers/net/ethernet/netswift/txgbe/txgbe_main.c
index 6be8d8cbbbff..cd7ae5703008 100644
--- a/drivers/net/ethernet/netswift/txgbe/txgbe_main.c
+++ b/drivers/net/ethernet/netswift/txgbe/txgbe_main.c
@@ -3705,7 +3705,11 @@ void txgbe_configure_isb(struct txgbe_adapter *adapter)
 
 	wr32(hw, TXGBE_PX_ISB_ADDR_L,
 			adapter->isb_dma & DMA_BIT_MASK(32));
+#ifdef CONFIG_64BIT
 	wr32(hw, TXGBE_PX_ISB_ADDR_H, adapter->isb_dma >> 32);
+#else
+	wr32(hw, TXGBE_PX_ISB_ADDR_H, 0);
+#endif
 }
 
 void txgbe_configure_port(struct txgbe_adapter *adapter)
-- 
2.33.0

